tests/ballerina-unit-test/src/test/resources/test-src/expressions/async/basic-async-operations.bal

```diff
@@ -61,8 +61,8 @@ function testAsyncNonNativeBasic7() returns int {
 }
 
 function testAsyncNonNativeBasic8() returns int {
-  future<int> f1 = start addNum(5, 2);
-  future<int> f2 = start addNum(5, 2);
+  future<int> f1 = start subtract(8, 3);
```

Comment:
do we have a test to see if the wait for any will return the successful future if only one future is canceled? I think there are tests for this kind of scenario when some panic happens in one future, I think we may need to add something like that for cancelation as well.

Reply:
Yes we do have a test https://github.com/ballerina-platform/ballerina-lang/pull/13090/files#diff-c57bc8d6b3a06c2b98c705942dcafed3R72. This will return the future that was not cancelled

Reply:
in that test, we cannot guarantee the canceled future will be really canceled right? (it may successfully finish execution, therefore test may intermittently fail) I think you need to use substract method there as well.